pkg/trace/api/otlp.go

```diff
@@ -305,12 +305,8 @@ func (o *OTLPReceiver) ReceiveResourceSpans(ctx context.Context, rspans ptrace.R
 			tagContainersTags: payloadTags.String(),
 		}
 	}
-	select {
-	case o.out <- &p:
-		// success
-	default:
-		log.Warn("Payload in channel full. Dropped 1 payload.")
-	}
+
+	o.out <- &p
```

Comment:
With this change we'll block the datadog exporter on `consumeTraces` (https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/d8a53842f0ed1074a5657c6fa015a55d831cd9e9/exporter/datadogexporter/traces_exporter.go#L128). Does the back pressure propagate all the way to the receivers?

@mx-psi Do you happen to know?

Reply:
I think it will end up being logged here https://github.com/open-telemetry/opentelemetry-collector/blob/a952082f73302cfa22880cc7afd933eb0fffac59/exporter/exporterhelper/common.go#L95-L96 and dropped, but I am not 100% sure

Reply:
Thanks, I took a look at exporterhelper and I think you're right. However exporterhelper can record and export metrics on dropped traces, that's still way better than the current behavior in trace-agent, i.e. dropping with only a warn log.